compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1873,6 +1873,66 @@ static FailureOr<Operation *> getRootOperation(
   return nullptr;
 }
 
+static LogicalResult adjustTileSizesForPackOp(func::FuncOp entryPointFn,
+                                              Operation *rootOp) {
+  // The check is not needed if we land
+  // https://github.com/openxla/iree/pull/12594
+  if (!rootOp) return success();
+
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(rootOp);
+  if (!linalgOp) return success();
+
+  // The transform dialect codegen has differnet logics and codegen flow. Ignore
+  // the adjustment for it.
+  auto pipeline = getTranslationInfo(entryPointFn).getPassPipeline().getValue();
+  if (pipeline == DispatchLoweringPassPipeline::TransformDialectCodegen) {
+    return success();
+  }
+  TileSizesListType tileSizesList =
+      getLoweringConfig(linalgOp).getTileSizeVals();
+
+  bool hasChanged = false;
+  auto res = entryPointFn.walk([&](tensor::PackOp packOp) -> WalkResult {
+    // Multiple pack ops case is not supported.
+    if (hasChanged) return WalkResult::interrupt();
+
+    // TODO(hanchung): Support chain cases. It needs following use-def chain
+    // until rootOp.
+    if (packOp.getSource().getDefiningOp() != rootOp) {
+      return WalkResult::advance();
+    }
+    hasChanged = true;
+    OpResult result = packOp.getSource().cast<OpResult>();
+    auto idxMap = linalgOp.getMatchingIndexingMap(
```

Comment:
I see why you are doing this. Maybe we can just change this to an assert, and say the pack op cannot be fused with a producer if the indexing map of the source in the producer is not identity.... We can enforce this logic in the fusion. Its kind of a half solution here cause you actually need to follow the indexing map from the source all the way upto the real root op to make sure you are doing this right.... We can just treat that as a contract that has to be satisfied if pack is to be fused with the producer.

Reply:
I think we'll hit multi-result generic op + pack op fusion soon; the results indexing maps might not all be identity?

Also, we'll have to handle reduction + elementwise + pack in the near future. We'll need similar things for tracking the dimensions.